### DISCUSSION TITLE
Difference between an Observation's StartTime and TimeTriggered?

### DISCUSSION THREAD

**Discussion by SRM256** [GridProtectionAlliance/gsf#82] [Q&A] [answered]
Both of these two properties are Timestamp types with absolute times. The help is unhelpful. One of these is the base time to use for relative times stored in the waveforms. The other is... I am not sure. In the IEEE spec document, the example always has the same time for both entries, so there's no disambiguation there that I can see. Any ideas?

        /// <summary>
        /// Gets the starting time of the data in the observation record.
        /// </summary>

        /// <summary>
        /// Gets the time the observation was triggered.
        /// </summary>

**Comment by StephenCWills** [comment, marked as answer]
One of the more obvious differences between these two timestamps is that the observation's start time is required whereas the trigger time is optional.

As you noted, the start time is used as the base time for relative timestamps within the descendant series instances. I've seen PQDIF files where the trigger time is encoded as the start time and the timestamps for series instances start out negative because the device captured pre-trigger data. In these cases, the trigger time can be the same as the start time or it can be omitted.

The key point here is that the start time does not have to be the same as the trigger time and can therefore be chosen more or less arbitrarily. For instance, you can choose to encode the start time as the timestamp of the first data point in the observation or the top of an interval in which the data was captured. The trigger time field is more well defined, essentially always representing the point in time at which the data source decided to capture some data in the PQDIF file.

Keep in mind that the standard only describes the structure and meaning of the encoded data. Even if the standard doesn't specify a particular meaning for the start time tag, PQDIF data processing and visualization tools may interpret the data differently depending on how you encode it. For example, I seem to recall that PQDiffractor uses the observation's start time as the x-axis origin when displaying data on a line chart. I'm not sure if it uses the trigger time for anything.
